components/esp_http_client/esp_http_client.c

```diff
@@ -297,6 +297,18 @@ esp_err_t esp_http_client_delete_header(esp_http_client_handle_t client, const c
     return http_header_delete(client->request->headers, key);
 }
 
+esp_err_t esp_http_client_get_username(esp_http_client_handle_t client, char **value)
```

Comment:
@hongquan May I request one more change in the form of validating `client` and `value` fields in both newly added public APIs (accordingly APIs can return `ESP_ERR_INVALID_ARG`)?

Reply:
Sure!